### PR TITLE
[Transform] Transform `_schedule_now` API

### LINKED ISSUE #44722: Add a "run_now" type of API for data frame transforms
For a continuously running data frame transform, the frequency could be set very high. However, if the user knows that they have new data that falls within the `delay` setting and want to see the updated results immediately, a `run_now` API would be useful. This will cause the current checkpoint to execute on the data set to gather the updated documents immediately. 

The next time the transform runs would still be up to the configured frequency.

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR introduces Transform `_schedule_now` API.

If you `_schedule_now` a transform, it will process the new data instantly, without waiting for the configured `frequency` interval.
After `_schedule_now` API is called, the transform will be processed again at `now + frequency` unless `_schedule_now` API is called again in the meantime.

Request is of the form:
```
POST _transform/{transform_id}/_schedule_now
```

If the transform `transform_id` does not exist, an error (`404`) is thrown.
If the transform exists but is stopped, `_schedule_now` reports success (`200`) but is effectively a no-op.

Closes https://github.com/elastic/elasticsearch/issues/44722

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                    if (e instanceof TransformTaskDisappearedDuringTriggerException) {
+                        logger.debug("[{}] transform task disappeared during trigger, ignoring", request.getId());
+                        listener.onResponse(Response.TRUE);
+                        return;
+                    }
+                    if (e instanceof TransformTaskTriggerException) {
+                        logger.warn(() -> format("[%s] failed to trigger running transform.", request.getId()), e);
+                        listener.onResponse(Response.TRUE);
+                        return;
+                    }
```
Comment (hendrikmuhs) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java:130]: in a mixed version cluster with old node versions you could get a `ActionNotFoundException`.

(alternatively you can filter/check the node version before `request.setNodes`, however `ActionNotFound` seems simpler, especially as this problem will eventually disappear)

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/TriggerTransformAction.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    private TriggerTransformAction() {
+        super(NAME, TriggerTransformAction.Response::new);
+    }
+
+    public static class Request extends BaseTasksRequest<Request> {
+
+        private final String id;
+
+        public Request(String id, TimeValue timeout) {
+            this.id = id;
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/TriggerTransformAction.java:42]: `ExceptionsHelper.requireNonNull(id, TransformField.ID.getPreferredName());` maybe?

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/action/TriggerTransformActionRequestTests.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+import org.elasticsearch.xpack.core.transform.action.TriggerTransformAction.Request;
+
+public class TriggerTransformActionRequestTests extends AbstractWireSerializingTestCase<Request> {
+
+    @Override
+    protected Request createTestInstance() {
+        return new Request(
+            randomBoolean() ? randomAlphaOfLengthBetween(1, 20) : Metadata.ALL,
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/action/TriggerTransformActionRequestTests.java:21]: I thought we don't allow `_all`?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/TriggerTransformAction.java (hunk 2 above)
Comment (hendrikmuhs) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/TriggerTransformAction.java:42]: timeout must be non-null it seems?

**Review by hendrikmuhs — COMMENTED** [review]
Nice!

I left some comments, I think mixed version needs to be handled otherwise this looks good to me.

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java (hunk 1 above)
Comment (hendrikmuhs) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java:102]: it should of course error. 

What error will be returned with letting it fall through?

Comment (przemekwitek) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java:102]: `POST _transform/test-from-1/_trigger`

errors the following way:
```
{
  "error": "no handler found for uri [/_transform/test-from-1/_trigger?pretty=true] and method [POST]"
}
```

This seems good to me.

Comment (przemekwitek) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java:102]: Actually I was too quick answering this comment.
FTR, I investigated more closely and here is what happens:
Setup:
2 nodes:
- 1 node (`localhost:9200`) running of `main` so it **doesn't** contain `_trigger` API
- 1 node (`localhost:9201`) running of this PR so it contains `_trigger` API
2 transforms:
- continuous transform `test-from-1` with 1 task on node `localhost:9201` (with `_trigger`)
- continuous transform `test-from-2` with 1 task on node `localhost:9200` (**without** `_trigger`)

Scenario:
```
elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9200/_transform/test-from-1/_trigger
{"error":"no handler found for uri [/_transform/test-from-1/_trigger] and method [POST]"}%                                                                                                                  

elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9200/_transform/test-from-2/_trigger
{"error":"no handler found for uri [/_transform/test-from-2/_trigger] and method [POST]"}%                                                                                                                  

elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9201/_transform/test-from-1/_trigger
{"acknowledged":true}%                                                                                                                                                                                      

elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9201/_transform/test-from-2/_trigger
{"acknowledged":true}%                                                                                                                                                                                      
```

The first 2 are obvious, we try to call the `_trigger` API on a node that doesn't have it, so the error is returned.
The 3. is ok. We call `_trigger` API on the node that has it, the task is on the same node so it works fine.
The 4. is not ok. We call `_trigger` API on the node that has it, but the task is on a different node.
The exception is swallowed, in the logs we see:
```
[2023-01-30T13:11:42,238][WARN ][o.e.x.t.a.TransportTriggerTransformAction] [node-trigger] [test-from-2] failed to trigger running transform.org.elasticsearch.xpack.transform.action.TransportTriggerTransformAction$TransformTaskTriggerException: Failed to trigger running transform.
	at org.elasticsearch.xpack.transform.action.TransportTriggerTransformAction.newResponse(TransportTriggerTransformAction.java:130)
	at org.elasticsearch.xpack.transform.action.TransportTriggerTransformAction.newResponse(TransportTriggerTransformAction.java:43)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction.newResponse(TransportTasksAction.java:214)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:325)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onFailure(TransportTasksAction.java:315)
```

Let me fix that...

Comment (przemekwitek) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java]: Ok, now the `ActionNotFoundTransportException` exception is a fall-through exception with the following API error message:
```
{
  "error": {
    "root_cause": [
      {
        "type": "action_not_found_transport_exception",
        "reason": "No handler for action [cluster:admin/transform/trigger[n]]"
      }
    ],
    "type": "action_not_found_transport_exception",
    "reason": "No handler for action [cluster:admin/transform/trigger[n]]"
  },
  "status": 500
}
```

Additionally, I made the `_trigger` API fail if the transform (config, **not** task) does not exist.

**Review by hendrikmuhs — APPROVED** [review]
LGTM! Great to see this implemented. 

FYI: @ajosh0504 @angorayc I think this is useful for your use cases as far as I remember our discussions.

**Comment by ajosh0504** [comment]
@hendrikmuhs OMG yes! For sure! Definitely going to check this out!

@rylnd @oatkiller FYI.

File: docs/changelog/92948.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+pr: 92948
+summary: Transform Trigger API
```
Comment (droberts195) [inline docs/changelog/92948.yaml:2]:
```suggestion
summary: Transform `_schedule_now` API
```

Comment (przemekwitek) [reply docs/changelog/92948.yaml]: Done, together with other renaming-related changes.

**Pull request merged**